### PR TITLE
CVCDiscrete extended to report correct MaxValue

### PR DESCRIPTION
**Pull request by cjakeman**
Optional parameter added to CVCDiscrete constructor so it can distinguish Lever|TwoState|TriState|MultiState and can set MaxValue correctly for use in CabControls API.

Specifically, the FRONT_HLIGHT is a tristate control (Off|Dim|Full) and was reported incorrectly.
See PR https://github.com/openrails/openrails/pull/377

Bug report: Missing (please add Elvas Tower, Trello, Launchpad link if one exists)

### CODE REVIEW

File: Source/Orts.Formats.Msts/CabViewFile.cs [hunk 1]
```diff
@@ -1149,14 +1157,23 @@ public CVCDiscrete(STFReader stf, string basepath)
...
-//                else                       // Unexpected error, so provide a hint
-//                    throw new STFException(stf, "Problem with NumPositions/NumValues/NumFrames/ScaleRange");
-//            } // End of Need check the Values collection for validity
+            //            catch (Exception error)
+            //            {
+            //                if (error is STFException) // Parsing error, so pass it on
+            //                    throw;
+            //                else                       // Unexpected error, so provide a hint
+            //                    throw new STFException(stf, "Problem with NumPositions/NumValues/NumFrames/ScaleRange");
+            //            } // End of Need check the Values collection for validity
```
Comment (Sharpe49) [inline Source/Orts.Formats.Msts/CabViewFile.cs:1176]: Don't keep commented code inside the repository.
Either remove it or put it inside preprocessor directives (#if DEBUG) if it is needed for debug purposes.

File: Source/Orts.Formats.Msts/CabViewFile.cs [hunk 2]
```diff
@@ -311,6 +311,14 @@ public enum CABViewControlUnits
         LBS
     }
 
+    public static class DiscreteStates
+    {
+        public const string Lever = "lever";
+        public const string TwoState = "twostate";
+        public const string TriState = "tristate";
+        public const string MultiState = "multistate";
+    }
```
Comment (Sharpe49) [inline Source/Orts.Formats.Msts/CabViewFile.cs:320]: I think an enumerated value would have been better here.
And we don't need to use this const string inside the first parameter TokenProcessor constructor below. It is less readable than before.

**Review by Sharpe49 — CHANGES_REQUESTED** [review]

File: Source/Orts.Formats.Msts/CabViewFile.cs (hunk 2 above)
Comment (cjakeman) [reply Source/Orts.Formats.Msts/CabViewFile.cs:322]: Thanks, Cedric. I'll try it like that.

File: Source/Orts.Formats.Msts/CabViewFile.cs (hunk 1 above)
Comment (cjakeman) [reply Source/Orts.Formats.Msts/CabViewFile.cs]: Agreed, but my intention is to fix a bug, not to review this class. That really requires a new PR.
I've now undone the format change that highlighted this section.

File: Source/Orts.Formats.Msts/CabViewFile.cs [hunk 3]
```diff
@@ -845,7 +855,7 @@ public class CVCDiscrete : CVCWithFrames
         private int numPositions;
         private bool canFill = true;
 
-        public CVCDiscrete(STFReader stf, string basepath)
+        public CVCDiscrete(STFReader stf, string basepath, DiscreteStates discreteStates)
```
Comment (Sharpe49) [inline Source/Orts.Formats.Msts/CabViewFile.cs:858]: The name of the parameter should be singular.

File: Source/Orts.Formats.Msts/CabViewFile.cs [hunk 4]
```diff
@@ -1293,8 +1312,8 @@ protected void ParseCustomParameters(STFReader stf)
     #region other controls
     public class CVCSignal : CVCDiscrete
     {
-        public CVCSignal(STFReader inf, string basepath)
-            : base(inf, basepath)
+        public CVCSignal(STFReader inf, string basepath, DiscreteStates discreteStates)
+            : base(inf, basepath, discreteStates)
```
Comment (Sharpe49) [inline Source/Orts.Formats.Msts/CabViewFile.cs:1316]: The name of the parameter should be singular.

**Review by Sharpe49 — CHANGES_REQUESTED** [review]
Seems good besides some parameter names that should be singular.

File: Source/Orts.Formats.Msts/CabViewFile.cs (hunk 3 above)
Comment (cjakeman) [reply Source/Orts.Formats.Msts/CabViewFile.cs]: Yes, of course it should.

Good catch.

File: Source/Orts.Formats.Msts/CabViewFile.cs (hunk 4 above)
Comment (cjakeman) [reply Source/Orts.Formats.Msts/CabViewFile.cs]: Yes, of course it should.

Good catch.

**Review by Sharpe49 — APPROVED** [review]

**Pull request merged**